src/plugins/intel_gpu/src/graph/border.cpp

```diff
@@ -65,6 +65,10 @@ std::vector<layout> border_inst::calc_output_layouts(border_node const& /*node*/
     };
 
     auto& memory_deps = impl_param.memory_deps;
+    if (desc->pad_value == 0 && (is_begin_mem || is_end_mem) && memory_deps.empty()) {
```

Comment:
I think we dont need to check value, 
but what we need to check is the memory dep count is equal to the required  # of memory deps.
i.e., 
if begin is from mem, end is constant => we need 2 mem deps 
if only begin is from mem, and end is constant => we need 1 mem dep 

So, please change it like 
if num_required_mem_dep < actual_mem_dep, return dynamic shape.